rllib/algorithms/appo/tf/appo_tf_learner.py

```diff
@@ -117,44 +119,41 @@ def compute_loss_per_module(
             ),
         )
 
-        action_kl = old_target_policy_dist.kl(target_policy_dist)
-        mean_kl_loss = tf.math.reduce_mean(action_kl)
+        if self.hps.use_kl_loss:
+            action_kl = old_target_policy_dist.kl(target_policy_dist)
+            mean_kl_loss = tf.math.reduce_mean(action_kl)
+        else:
+            mean_kl_loss = 0.0
         mean_pi_loss = -tf.math.reduce_mean(surrogate_loss)
 
         # The baseline loss.
         delta = values_time_major - vtrace_adjusted_target_values
         mean_vf_loss = 0.5 * tf.math.reduce_mean(delta**2)
 
         # The entropy loss.
-        mean_entropy_loss = -tf.math.reduce_mean(target_actions_logp_time_major)
+        mean_entropy_loss = -tf.math.reduce_mean(target_policy_dist.entropy())
```

Comment:
oh the reason for using `target_actions_logp_time_major` is because in the past ive found that the analytical entropy `(-mean (logp(x)))` is better than using `entropy()`. Although it looks like this right over here was the actual bug.

Because we were adding the entropy to the loss before instead of subtracting it, we were doing the exact opposite of max entropy RL.

I'd bet that this was the reason why our setup broke.

Reply:
Let's leave it as is for now (`entropy()`). Happy to later try the correct negative `-mean (logp(x))`, but we should always confirm with our hard-task benchmarks first and right now, these are working fine with `entropy()`.

Reply:
yeah either will work. The most important thing here is adding that additional minus sign.